### ISSUE TITLE
Websocket not picked up after connection error

### ISSUE THREAD

**Issue by chris-aeviator** [gotify/server#96]
I have the situation that gotify works reliably when I reload it's web-ui or android app, but will not receive anymore notifications after the websocket connection has been "disturbed" (which happens quite often). I can confirm that other websocket connections on the same server run without any issue.

It seems like the websocket connection is not reliably re-established. I'm happy to provide further (debug) information if pointed into a direction to debug.

**Comment by jmattheis** [comment]
Do you use a reverse proxy like nginx? Which Browser do you use?
Could you have a look at your Gotify Android-App log (it is the last option in the drawer in the app). 
The WebSocket status and reconnect attempts will be logged there with messages like: `WebSocket: opened` or `WebSocket: failure`.

**Comment by chris-aeviator** [comment]
@jmattheis thx for your quick feedback!

I'm using an nginx-reverse proxy on the server. Here's my Gotify Android log (as screenshot :) )

![image](https://user-images.githubusercontent.com/11522213/49748181-4a3b3e00-fca5-11e8-9349-c5c2f9804205.png)

I'm running on Mozilla Firefox 62.0.3 (Arch) on the Desktop & can replicate the behaviour by letting the Gotify UI rest for some minutes in an not active tab. A project running with the same nginx config, that's using websockets, will display notifications for an (inactive) tab in my browser.

**Comment by jmattheis** [comment]
@chris-aeviator Could you show your nginx config? I've left a firefox tab with Gotify open for around 2 hours (on Windows) and it still works as expected.

The multiple "WebSocket opened"-messages in the log are probably a bug, could you try to force stop the Gotify app? Have you disabled the battery optimization for the Gotify android app?

**Comment by jmattheis** [comment]
I'm closing this for now because I can't reproduce the problem.

**Comment by dugite-code** [comment]
I've encountered the same error and believe I have found myself solution. It looks like it might have something to do with the keep `proxy_read_timeout` and `proxy_send_timeout`. I've changed it to a whole day but I plan to crank it down to test where it actually needs to be.

My nginx config:
```
upstream gotify {
  keepalive 60;
  server 127.0.0.1:9980;
}


  location = /gotify {
        return 301 /gotify/;
  }

server {
  listen 443 ssl http2;

[...]

  location ^~ /gotify/ {
        rewrite ^/gotify/(.*) /$1  break;
        proxy_redirect ~^/(.*)$ /gotify/$1;
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Real-IP $remote_addr;

        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $http_connection;
        proxy_buffering off;

        proxy_read_timeout 86400s;
        proxy_send_timeout 86400s;
        proxy_pass http://gotify$uri$is_args$args;
  }

  location ^~ /gotify/stream {
        rewrite ^/gotify/(.*) /$1  break;
        access_log /var/log/nginx/access.log;

        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host $host;

        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $http_connection;
        proxy_buffering off;

        proxy_read_timeout 86400s;
        proxy_send_timeout 86400s;

        proxy_pass http://gotify$uri$is_args$args;
  }
}
```

**Comment by jmattheis** [comment]
See also https://github.com/gotify/server/issues/82 in there `proxy_read_timeout` and `proxy_send_timeout` where mentioned too.

**Comment by Akito13** [comment]
> 
> 
> See also #82 in there `proxy_read_timeout` and `proxy_send_timeout` where mentioned too.

I am not sure if my issue is related to this, but here we go. When I connect to the Gotify Web-UI within Firefox on my computer, I constantly get an error in the bottom right corner saying _WebSocket connection closed, trying again in 30 seconds_. The thing is:
1. I don't use a reverse proxy, I connect directly to the server.
2. Everything is working fine, I don't have any issues with the service itself. The only thing that is bugging me is this (unnecessary?) message appearing every 30 seconds. I wonder if there is an actual problem behind the scenes. At the very least, this message constantly is appearing, which is a problem.

**Comment by eternal-flame-AD** [comment]
@Akito13 Have you upgraded your server to v2.0.3? We have recently fixed an issue( #150  ) which might cause this problem.

**Comment by Akito13** [comment]
> 
> 
> @Akito13 Have you upgraded your server to v2.0.3? We have recently fixed an issue( #150 ) which might cause this problem.

Indeed, this solves it, thank you.
